inc/item.class.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/**
+ * FusionInventory
+ *
+ * Copyright (C) 2010-2016 by the FusionInventory Development Team.
+ *
+ * http://www.fusioninventory.org/
+ * https://github.com/fusioninventory/fusioninventory-for-glpi
+ * http://forge.fusioninventory.org/
+ *
+ * ------------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of FusionInventory project.
+ *
+ * FusionInventory is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * FusionInventory is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with FusionInventory. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * ------------------------------------------------------------------------
+ *
+ * This file is used to manage the extended information of a computer.
+ *
+ * ------------------------------------------------------------------------
+ *
+ * @package   FusionInventory
+ * @author    David Durieux
+ * @copyright Copyright (c) 2010-2016 FusionInventory team
+ * @license   AGPL License 3.0 or (at your option) any later version
+ *            http://www.gnu.org/licenses/agpl-3.0-standalone.html
+ * @link      http://www.fusioninventory.org/
+ * @link      https://github.com/fusioninventory/fusioninventory-for-glpi
+ *
+ */
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access directly to this file");
+}
+
+/**
+ * Common class to manage informations FI display on an asset
+ */
+class PluginFusioninventoryItem extends CommonDBTM {
+
+   /**
+    * The right name for this class
+    *
+    * @var string
+    */
+   static $rightname = '';
+
+   /**
+   * The itemtype we are processing
+   */
+   public $itemtype  = '';
+
+   /**
+   * Get automatic inventory info for a computer
+   * @since 9.1+1.2
+   * @param $items_id the item ID to look for
+   * @return inventory computer infos or an empty array
+   */
+   function hasAutomaticInventory($items_id) {
+      $fk     = getForeignKeyFieldForItemType($this->itemtype);
+      $params = [$fk => $items_id];
+      if ($this->getFromDBByCrit($params)) {
+         return $this->fields;
+      } else {
+         return [];
+      }
+   }
+
+   /**
+   * Form to download the XML inventory file
+   *
+   * @param $items_id the item ID to look for
```

Comment:
Param type is missing
Return is missing as well

Reply:
done